src/reducers/index.js

```diff
@@ -511,17 +522,25 @@ export const resetSelectedTokenIfNeeded = (state, action) => {
   return state;
 };
 
-/*
+/**
  * Used when registering or creating tokens to update the wallet token list.
-*/
+ * @param {Record<string,string>} state.allTokens - Current list of allTokens
+ * @param {string} action.payload.uid - UID of token to add
+ */
 export const onNewTokens = (state, action) => {
+  // Convert `allTokens` to a Set to prevent duplicates
+  const allTokensSet = new Set();
+  for (const token of Object.keys(state.allTokens)) {
+    allTokensSet.add(token);
+  }
   // Add new created token to the all tokens set
-  state.allTokens.add(action.payload.uid);
+  allTokensSet.add(action.payload.uid);
 
   return {
     ...state,
     selectedToken: action.payload.uid,
     tokens: action.payload.tokens,
+    allTokensSet: helpersUtils.convertSetToObject(allTokensSet),
```

Comment:
Should the key name be `allTokensSet`?

Reply:
No, it shouldn't! It was certainly a leftover from some experimentation and was broken as it was.
Fixed on https://github.com/HathorNetwork/hathor-wallet/pull/587/commits/3ee5ac5cfba3a1e7dbe5387109159d06d4a34485